app/serializers/cache/footnote_serializer.rb

```diff
@@ -48,8 +43,34 @@ def as_json
           }
         end
       end
-      footnote_attributes[:extra_large_measures] = extra_large_measures
+      footnote_attributes[:extra_large_measures] = extra_large_measures?
       footnote_attributes
     end
+
+    private
+
+    def goods_nomenclatures
+      @goods_nomenclatures ||= footnote.goods_nomenclatures.compact.select do |goods_nomenclature|
+        has_valid_dates(goods_nomenclature) &&
+          HiddenGoodsNomenclature.codes.exclude?(goods_nomenclature.goods_nomenclature_item_id)
+      end
+    end
+
+    def measures
+      @measures ||= footnote
+        .measures_dataset
+        .eager(:goods_nomenclature)
+        .exclude(goods_nomenclature_item_id: nil)
+        .all
```

Comment:
we might want to move the first part of this, on the Footnote model

```
= footnote
        .measures_dataset
        .eager(:goods_nomenclature)
        .exclude(goods_nomenclature_item_id: nil)
        .all
```

Reply:
I think probably we could rejig all of the measure datasets to exclude the dead nomenclature and DRY these things up a bit/make the rest of the application safer.